src/utils/dex/evolution.js

```diff
@@ -235,6 +236,187 @@ function validIds(path, ar) {
   });
 }
 
+function genForms() {
+  const formNamedata = formPokemonNames;
+  const formsList = formNamedata["labelDataArray"];
+  const forms = {};
+
+  for (let i = 0; i < formsList.length; i++) {
+    const allForms = formsList[i];
+    if (allForms["arrayIndex"] !== 0 && parseInt(allForms["labelName"].slice(-3)) > 0) {
+      forms[allForms["labelName"]] = allForms["arrayIndex"];
+    }
+  }
+
+  return forms;
+}
+
+const forms = genForms();
+
+function getFormFormat(monsNo, formNo) {
+  const monZeros = 3 - String(monsNo).length;
+  const formZeros = 3 - String(formNo).length;
+  if (String(monsNo).length > 3) {
+    return `ZKN_FORM_$${monsNo}_${'0'.repeat(formZeros)}${formNo}`;
+  }
+  return `ZKN_FORM_${'0'.repeat(monZeros)}${monsNo}_${'0'.repeat(formZeros)}${formNo}`;
+}
+
+function removeDuplicates(pathDictionary) {
+  const newPath = [];
+  for (const pathElement of pathDictionary) {
+    if (!newPath.includes(pathElement)) {
+      newPath.push(pathElement);
+    }
+  }
+  return newPath;
+}
+
+function removeDuplicateForms(evolutionPaths) {
```

Comment:
If you want to ensure uniqueness, you can use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) instead